File: app/src/main/java/com/example/aac/java/TestPresenter.java
package com.example.aac.java;

import android.arch.lifecycle.LiveData;
import android.util.Log;

import com.aac.data.http.converter.BeanConverter;
import com.aac.data.http.utils.AacUtils;
import com.aac.module.ui.AacPresenter;
import com.alibaba.fastjson.TypeReference;
import com.lzy.okgo.model.HttpParams;

import java.lang.reflect.Type;
import java.util.List;

import io.reactivex.Flowable;

/**
 * Created by yangc on 2017/8/13.
 * E-Mail:yangchaojiang@outlook.com
 * Deprecated:
 */

public class TestPresenter extends AacPresenter<MainActivity> {
    public static final String TAG = "TestPresenter";


    @Override
    protected void onCreateView() {
        super.onCreateView();
        Log.d(TAG, "onCreateView");
    }

    public LiveData<String> sss() {
        HttpParams httpParams = new HttpParams();
       return AacUtils.httpLiveGet("sss"
                , httpParams
                , new BeanConverter<String>("-1", String.class));
    }

    public Flowable<List<String>> sss2() {
        HttpParams params = new HttpParams();
        TypeReference typeReference = new TypeReference<List<String>>(){};
        return AacUtils.httpRxGet("sss"
                , params
                , new BeanConverter<>("-1",typeReference.getType()));
    }


}

File: app/src/main/java/com/example/aac/java/data/test/presenter/TestPresenter.kt
package com.example.aac.java.data.test.presenter

import com.aac.expansion.list.AacMultiListAPresenter

import com.example.aac.java.data.test.model.TestViewModel
import com.example.aac.java.data.test.ui.TestActivity
import com.example.aac.java.data.test.bean.Test;


/**
* Created by yangc on 2018/10/15 18:14:55
* E-Mail:yangchaojiang@outlook.com
* Deprecated:
**/
class TestPresenter : AacMultiListAPresenter<TestActivity, Test> (){
    private lateinit var mTest: TestViewModel
    public override fun onCreate() {
        super.onCreate()
        mTest = getViewModel(TestViewModel::class.java)
     }

 fun lazyLoad() {
      }
    companion object {
        val TAG = TestPresenter::class.java.name
    }
}
